server/reload.go

```diff
@@ -1100,6 +1133,33 @@ func (s *Server) reloadClientTraceLevel() {
 	}
 }
 
+// reloadClusterName detects cluster membership changes triggered
+// due to a reload where the name changes.
+func (s *Server) reloadClusterName() {
+	var (
+		routesa [64]*client
+		routes  = routesa[:0]
+	)
+
+	s.mu.Lock()
+
+	// Get all connected routes.
+	for _, route := range s.routes {
+		route.mu.Lock()
+		isClusterMember := route.clusterName == s.info.Cluster
+		if !isClusterMember {
+			routes = append(routes, route)
+		}
+		route.mu.Unlock()
+	}
+	s.mu.Unlock()
+
+	// Give up the lock before starting to disconnect.
+	for _, route := range routes {
+		route.closeConnection(ClusterNameConflict)
```

Comment:
Should we disconnect or send an INFO update? Then all of the logic should just work like on first connect.

Reply:
Changed to send INFO instead of disconnecting.